### DISCUSSION TITLE
getOrNull() method in addition to exists()

### DISCUSSION THREAD

**Discussion by V-ctor** [selenide/selenide#2134] [Ideas]
Hi! 
Do we have any way to get some element or null (if it does not exist)? Some getOrNull() method like exists()? I could not find.
But I'm sure that it could be useful to write code like this (Kotlin):
`find(...).getOrNull()?.value`
without exception handling or checking by exists() method.
What would you say?

**Comment by asolntsev** [comment]
Yes, Selenide doesn't have such a method.
And I don't think it's a good idea to use such constructions in tests.
Test is a specification, it has to _specify_ that the element should exist. Or should not exist. 
Using "ifs" in tests is evil, isn't it?

**Reply by V-ctor** [reply]
But in the same time we have exists() method. Is not it about "ifs"? getOrNull is just the same thing.
And to be honest yes I use Selenide not only for complete auto test but for semi-manual/auto test. Let's say for some scripts that help to pass through boring pages like login if my session is expired or sort of then it could be useful to write like this:
```
find("input.login_name").getOrNull()?.value=loginName
find("input.password").getOrNull()?.value=password
find("button.login").getOrNull()?.click()

```
Login page is just simple example it can be more complicated  domain/business specific things when we have to use manual testing but we can avoid some annoying repeating steps.

**Reply by asolntsev** [reply]
Yes, `SelenideElement` has method `exists()` (as well as `getText()`, `isDisplayed()`, `isEnabled()`) - and I regret about it. 
Such methods facilitate IFs appearing in tests. :(

In your example, I think using `getOrNull()` for three elements is not a good idea. 

I would rather check only once that the session has disappeared:
```java
if (sessionExpired()) {
  find("input.login_name").value=loginName
  find("input.password").value=password
  find("button.login").click()
}
```
Ideally, `sessionExpired()` should not check UI elements, but sessionStorage, cookies or something else. Or even API call. 

Furthermore, I recommend to initialize a session in the beginning of every test using some stable mechanism, not UI. For example, API call.
